### PR TITLE
Upgrade to go1.18.6 (7.39.x)

### PR DESCRIPTION
**Pull request by djmitche**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .github/workflows/go_mod_tidy.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
     - name: Install go
       uses: actions/setup-go@v2
       with:
-        go-version: '1.17.6'
+        go-version: '1.18.6
```
Comment (sgnn7) [inline .github/workflows/go_mod_tidy.yml:31]:
```suggestion
        go-version: '1.18.6'
```

File: .github/workflows/windows-unittests.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Install go
         uses: actions/setup-go@v2
         with:
-          go-version: '1.17.11'
+          go-version: '1.18.6
```
Comment (sgnn7) [inline .github/workflows/windows-unittests.yml:28]:
```suggestion
          go-version: '1.18.6'
```

File: releasenotes/notes/go118-b3ac691999dd0015.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+security:
+  - |
+    The Agent is built with go1.18.6.
```
Comment (sgnn7) [inline releasenotes/notes/go118-b3ac691999dd0015.yaml:11]: Should we include the extra comment here from the other PR around the mem tweaking here as well?

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@djmitche Couple of nits and a comment

**Review by sgnn7 — APPROVED** [review]

**Review by kayayarai — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
For the `dependencies` CircleCI failing job I would suggest doing the same as in the builders repo: modifying the `requirements.txt` file https://github.com/DataDog/datadog-agent/blob/a4f7ee649e2d6c1830d67e56f286310d5f561fc9/requirements.txt#L2 so that it points to `7.39.x` instead of `main`, to ensure that the requirements used are the same as in the 7.39.0 build.

**Pull request closed without merging**